### ISSUE TITLE
RtmpClient unable to properly read stream

### ISSUE THREAD

**Issue by ojw28** [ant-media/LibRtmp-Client-for-Android#70]
This issue was originally reported here: https://github.com/google/ExoPlayer/issues/4319

It appears RtmpClient is unable to read `rtmp://76.81.29.11:1935/live/Cam2` correctly. To verify the issue is in RtmpClient I did the following:

* Capture a dump of the stream using `rtmpdump` (command line tool). Side load this file onto an Android device and verify it can be played correctly using ExoPlayer.
* Capture a dump of the stream using LibRtmp (on device). Try and play this. It cannot be played correctly, and shows the same behavior as when trying to play the stream directly in ExoPlayer.

**Comment by Srinu-try** [comment]
Hi,

I am facing the above issue with librtmp library used in ExoPlayer.

Request your help in this regard.

**Comment by woehrl** [comment]
Hi,

seems like this problem is still not fixed, right? I have a HD video stream (rtmp) without any audio stream and it just does not play back. If I add a nullsrc audio then it works, but that adds more processing power and time that I don't have available.
The stream looks like this: https://a.uguu.se/Oj8rdrFDO7v3_rtmp_h264_noaudio.flv

Requesting help! Best regards...

**Comment by ojw28** [comment]
The link you've provided above plays fine for me. It's also an https link rather than rtmp?

**Comment by woehrl** [comment]
Thank you for your swift reply! yes, the file plays perfectly in media players (MPC, VLC). I stored the rtmp stream to a file (raw conversion in VLC) for you to have a look at the stream - what might be wrong with it. I'm using librtmp within ExoPlayer on Android, already using latest release 3.1.0. I could provide you a direct link to the stream source via PN.

**Comment by umangsaini** [comment]
Few FLV files signal that both A and V tracks are available, when in fact only Video is available. This creates A/V tracks which will not play if Audio samples are not present.

Can you check in 
Exoplayer **FlvExtractor.java**  -> readFlvHeader() if 
`boolean hasAudio = (flags & 0x04) != 0;  `

If **hasAudio** flag is set true for your FLV file, then Video only playback will be an issue. FFMPEG is able to play such content over RTMP.

**Comment by woehrl** [comment]
Ok, I managed to clone the repo and included it in my project. That way I could perfectly detect if the flv headers set an audio track.

I found out that my gstreamer pipeline including the flvmux element did NOT add an audio stream, but the receiving nginx-rtmp module by default sets an audio track as given even if there is none in the input stream: [url](https://github.com/arut/nginx-rtmp-module/issues/641)

So I just had to set `meta copy;` in my nginx server configuration. Now it works!

Thank you very much for your input! Best regards, Florian
